Melosys 5065

ivanskodje opened the pull request:
Ref: https://jira.adeo.no/browse/MELOSYS-5065

**NB: Denne krever at melosys-api er oppdatert først**

Lagt til erHistorikk og erHistorikk for å kunne filtrere bort historisk data. 
Det ble valgt å endre graphql query ved å legge til erHistorikk direkte inni familiemedlemmer, istedet for å legge til et underobjekt som ikke er nødvendig. 


> Hva hvis vi trenger mer data i fremtiden? 

YAGNI

ivanskodje committed:
Oppdatert familiemedlemmer til å inkludere en nødvendig erHistorikk for å kunne filtrere bort utdatert data

ivanskodje committed:
Merge branch 'master' into MELOSYS-5065

martonSkjevelandNav commented on src/graphql/generated/types.ts:
```diff
@@ -36,6 +36,7 @@ export type Familiemedlem = {
   foreldreansvar?: Maybe<Scalars['String']>;
   fnrAnnenForelder?: Maybe<Scalars['String']>;
   sivilstand?: Maybe<Scalars['String']>;
+  erHistorisk?: Maybe<Scalars['Boolean']>;
```
`erHistorisk` kan vel ikke være null? Samme felt for andre pdl queries kan iaf ikke det

martonSkjevelandNav replied:
Trenger isåfall heller ikke sjekken som blir gjort i familiemedlemmer.tsx

ivanskodje replied:
Så vidt jeg forstår det har ikke alle familiemedlemmer en sivilstand (gift, ugift, separert), som eks. barn. Eller kanskje det gjør det? @martonSkjevelandNav

martonSkjevelandNav replied:
Det stemmer med sivilstand ja: den kan være null. Men `erHistorisk` vil alltid ha en verdi: `true` eller `false`

ivanskodje replied:
I frontend mener du? I backend sender vi null hvis det ikke er sivilstand. 
Derimot så løses dette når jeg har gjort en endring i backend til å sette sivilstand objektet (som inneholder erHistorisk) istedet, ny commit kommer :)

ivanskodje replied:
Løst ved å ta vekk erHistorisk og istedet eksponere sivilstand objektet

ivanskodje committed:
Merge remote-tracking branch 'origin/master' into MELOSYS-5065

ivanskodje committed:
Endre hentFamiliemedlemmer sin graphql til å hente sivilstand objekt.

Erstattet bruk av gamle sivilstand med sivilstand.type, og gamle sivilstandGyldighetsperiodeFom med sivilstand.gyldigFraOgMed.

Gjort om sivilstand type definisjon på sivilstand.master til optional ettersom det bryter med graphql fleskbilitet  om å kunne spørre om data vi selv ønsker, uten å måtte legge til unødvendig data.

martonSkjevelandNav commented on src/graphql/generated/getSdk.ts:
```diff
@@ -157,7 +156,7 @@ type Sivilstand = {
   gyldigFraOgMed?: Maybe<Scalars['Date']>;
   bekreftelsesdato?: Maybe<Scalars['Date']>;
-  master: Scalars['String'];
+  master?: Maybe<Scalars['String']>;
```
Denne ble vel endret tilbake i api, så må generate på nytt

martonSkjevelandNav reviewed (approved):
Kjør generate graphql på nytt etter endring i api, så ser det good ut 👍

ivanskodje committed:
Merge remote-tracking branch 'origin/master' into MELOSYS-5065

ivanskodje committed:
Har lagt til sivilstand.master ettersom det er et krav som følge av generering av typer (not nullable)

ivanskodje committed:
Oppdatert test til å ta i bruk master (som må være med pga. type har non nullable på master feltet)

martonSkjevelandNav reviewed (approved):
🚀
